src/videojs.ads.js

```diff
@@ -477,11 +460,33 @@ var
             }
           },
           'content-playback': {
+            enter: function() {
+              player.trigger({
+                type: 'contentplayback',
+                triggerevent: fsm.triggerevent
+              });
+              // Make sure that the cancelPlayTimeout is cleared
+              if (player.ads.cancelPlayTimeout) {
+                clearImmediate(player.ads.cancelPlayTimeout);
+                player.ads.cancelPlayTimeout = null;
+                // Trigger playback if play was canceled and the player is paused.
+                if (player.paused()) {
+                  player.play();
+                }
+              }
+            },
             events: {
+              // in the case of a timeout, adsready might come in late.
+              'adsready': function() {
+                if (player.paused()) {
+                  this.state = 'ads-ready';
+                } else {
+                  this.state = 'preroll?';
```

Comment:
Yes, in my investigation the state of the player, etc would be the same here as it would have been in ad-timeout-playback state and having listening for late arriving adsready calls makes sense the same here as it would have there.